Errors during build from master

Just forked the repo and tried to build from the unchanged sources.
Using Debian/testing, python 2.7, cython 15.1, libzmq master aka 3.3
_LOTS_ of errors on cython compile.

Installing egg from pypi works though.

No, can't use the egg because I want to modify pyzmq.

Cheers
  Guido

Output:

```
running build_ext
running configure
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -c build/temp.linux-x86_64-2.7/scratch/check_sys_un.c -o build/temp.linux-x86_64-2.7/scratch/check_sys_un.o
gcc -pthread build/temp.linux-x86_64-2.7/scratch/check_sys_un.o -lzmq -o build/temp.linux-x86_64-2.7/scratch/check_sys_un
************************************************
Configure: Autodetecting ZMQ settings...
    Custom ZMQ dir:       
cc -c /home/guido/tmp/timer_createS5r7Pw.c -o home/guido/tmp/timer_createS5r7Pw.o
cc home/guido/tmp/timer_createS5r7Pw.o -o a.out
home/guido/tmp/timer_createS5r7Pw.o: In function `main':
timer_createS5r7Pw.c:(.text+0x15): undefined reference to `timer_create'
collect2: error: ld returned 1 exit status
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -Izmq/utils -Izmq/core -Izmq/devices -c build/temp.linux-x86_64-2.7/scratch/vers.c -o build/temp.linux-x86_64-2.7/scratch/vers.o
gcc -pthread build/temp.linux-x86_64-2.7/scratch/vers.o -lzmq -lrt -o build/temp.linux-x86_64-2.7/scratch/vers
    ZMQ version detected: 3.3.0
************************************************
skipping 'zmq/core/_device.c' Cython extension (up-to-date)
building 'zmq.core._device' extension
creating build/temp.linux-x86_64-2.7/zmq
creating build/temp.linux-x86_64-2.7/zmq/core
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_SYS_UN_H=1 -Izmq/utils -Izmq/core -Izmq/devices -I/usr/include/python2.7 -c zmq/core/_device.c -o build/temp.linux-x86_64-2.7/zmq/core/_device.o
gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro build/temp.linux-x86_64-2.7/zmq/core/_device.o -lzmq -lrt -o build/lib.linux-x86_64-2.7/zmq/core/_device.so
skipping 'zmq/core/_poll.c' Cython extension (up-to-date)
building 'zmq.core._poll' extension
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_SYS_UN_H=1 -Izmq/utils -Izmq/core -Izmq/devices -I/usr/include/python2.7 -c zmq/core/_poll.c -o build/temp.linux-x86_64-2.7/zmq/core/_poll.o
gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro build/temp.linux-x86_64-2.7/zmq/core/_poll.o -lzmq -lrt -o build/lib.linux-x86_64-2.7/zmq/core/_poll.so
skipping 'zmq/core/_version.c' Cython extension (up-to-date)
building 'zmq.core._version' extension
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DHAVE_SYS_UN_H=1 -Izmq/utils -Izmq/core -Izmq/devices -I/usr/include/python2.7 -c zmq/core/_version.c -o build/temp.linux-x86_64-2.7/zmq/core/_version.o
gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro build/temp.linux-x86_64-2.7/zmq/core/_version.o -lzmq -lrt -o build/lib.linux-x86_64-2.7/zmq/core/_version.so
cythoning zmq/core/constants.pyx to zmq/core/constants.c

Error compiling Cython file:
------------------------------------------------------------
...

#-----------------------------------------------------------------------------
# Symbols to export
#-----------------------------------------------------------------------------

__all__ = [ key for key in locals().keys() if not key.startswith('_') ]
                                ^
------------------------------------------------------------

zmq/core/constants.pyx:190:33: Cannot convert 'size_t (zmq_msg_t *) nogil' to Python object

Error compiling Cython file:
------------------------------------------------------------
...

#-----------------------------------------------------------------------------
# Symbols to export
#-----------------------------------------------------------------------------

__all__ = [ key for key in locals().keys() if not key.startswith('_') ]
                                ^
------------------------------------------------------------

zmq/core/constants.pyx:190:33: Cannot convert 'int (void *) nogil' to Python object

Error compiling Cython file:
------------------------------------------------------------
...

#-----------------------------------------------------------------------------
# Symbols to export
#-----------------------------------------------------------------------------

__all__ = [ key for key in locals().keys() if not key.startswith('_') ]
                                ^
------------------------------------------------------------

zmq/core/constants.pyx:190:33: Cannot convert 'char *(int) nogil' to Python object

Error compiling Cython file:
------------------------------------------------------------
...

#-----------------------------------------------------------------------------
# Symbols to export
#-----------------------------------------------------------------------------

__all__ = [ key for key in locals().keys() if not key.startswith('_') ]
                                ^
------------------------------------------------------------

zmq/core/constants.pyx:190:33: Cannot convert 'int (zmq_msg_t *, zmq_msg_t *) nogil' to Python object

Error compiling Cython file:
------------------------------------------------------------
...

#-----------------------------------------------------------------------------
# Symbols to export
#-----------------------------------------------------------------------------

__all__ = [ key for key in locals().keys() if not key.startswith('_') ]
                                ^
------------------------------------------------------------

```

--- a/README.md
+++ b/README.md
@@ -50,7 +50,7 @@ the PyZMQ source code, eggs, or MSI installer from
 
 You can also get the latest source code from our GitHub repository, but
 building from the repository will require that you install Cython
-version 0.15 or later.
+version 0.16 or later.
 
 ## Building and installation
 

--- a/setup.py
+++ b/setup.py
@@ -47,6 +47,7 @@ from distutils.errors import CompileError, LinkError
 from distutils.command.build import build
 from distutils.command.build_ext import build_ext
 from distutils.command.sdist import sdist
+from distutils.version import LooseVersion as V
 
 from unittest import TextTestRunner, TestLoader
 from glob import glob
@@ -755,14 +756,14 @@ class CheckSDist(sdist):
         sdist.run(self)
 
 class CheckingBuildExt(build_ext):
-    """Subclass build_ext to get clearer report if Cython is neccessary."""
+    """Subclass build_ext to get clearer report if Cython is necessary."""
     
     def check_cython_extensions(self, extensions):
         for ext in extensions:
           for src in ext.sources:
             if not os.path.exists(src):
                 fatal("""Cython-generated file '%s' not found.
-                Cython is required to compile pyzmq from a development branch.
+                Cython >= 0.16 is required to compile pyzmq from a development branch.
                 Please install Cython or download a release package of pyzmq.
                 """%src)
     
@@ -829,12 +830,35 @@ submodules = dict(
 )
 
 try:
+    import Cython
+    if V(Cython.__version__) < V('0.16'):
+        raise ImportError("Cython >= 0.16 required, have %s" % Cython.__version__)
     from Cython.Distutils import build_ext as build_ext_c
     cython=True
 except ImportError:
     cython=False
     suffix = '.c'
     cmdclass['build_ext'] = CheckingBuildExt
+    
+    class MissingCython(Command):
+        
+        user_options = []
+        
+        def initialize_options(self):
+            pass
+        
+        def finalize_options(self):
+            pass
+        
+        def run(self):
+            try:
+                import Cython
+            except ImportError:
+                warn("Cython is missing")
+            else:
+                if V(Cython.__version__) < V('0.16'):
+                    warn("Cython >= 0.16 required for compiling Cython sources, but we have %s" % Cython.__version__)
+    cmdclass['cython'] = MissingCython
 else:
     
     suffix = '.pyx'

